ansible/configs/ocp4-cluster/pre_infra.yml

```diff
@@ -56,13 +56,23 @@
       when: agnosticd_aws_capacity_reservation_results.reservations | default({}) | length > 0
       block:
       - name: Set availability zone for bastion, masters and workers
+        vars:
+          _r: "{{ agnosticd_aws_capacity_reservation_results.reservations }}"
         set_fact:
           # Cloudformation (bastion)
-          aws_availability_zone: "{{ agnosticd_aws_capacity_reservation_results.reservations.az1.availability_zone }}"
+          aws_availability_zone: "{{ _r.az1.availability_zone }}"
           # masters
+          # Support several logical mapping (masters, masters1-3)
           openshift_controlplane_aws_zones_odcr:
-          - "{{ agnosticd_aws_capacity_reservation_results.reservations.masters.availability_zone }}"
+          - "{{ _r.masters.availability_zone  | default(None) }}"
+          - "{{ _r.masters1.availability_zone | default(None) }}"
+          - "{{ _r.masters2.availability_zone | default(None) }}"
+          - "{{ _r.masters3.availability_zone | default(None) }}"
```

Comment:
It seems a bit awkward to have these `None` values here? Perhaps instead moving the `select` here?
```
      openshift_controlplane_aws_zones_odcr: >-
        {{ [
          _r.masters.availability_zone | default(None),
          _r.masters1.availability_zone | default(None),
          _r.masters2.availability_zone | default(None),
          _r.masters3.availability_zone | default(None),
        ] | select() | list }}
```

Reply:
yeah i thought of that but i thought it was even more weird?